browser/ui/sidebar/sidebar_web_panel_controller.h

```diff
@@ -7,32 +7,45 @@
 #define BRAVE_BROWSER_UI_SIDEBAR_SIDEBAR_WEB_PANEL_CONTROLLER_H_
 
 #include "base/memory/raw_ref.h"
+#include "brave/components/sidebar/browser/sidebar_item.h"
+#include "chrome/browser/ui/tabs/tab_strip_model_observer.h"
 
 class BrowserView;
 class BraveMultiContentsView;
 
-namespace sidebar {
+namespace content {
+class WebContents;
+}  // namespace content
 
-struct SidebarItem;
+namespace sidebar {
 
-class SidebarWebPanelController {
+class SidebarWebPanelController : public TabStripModelObserver {
  public:
   explicit SidebarWebPanelController(BrowserView& browser);
-  ~SidebarWebPanelController();
+  ~SidebarWebPanelController() override;
 
   SidebarWebPanelController(const SidebarWebPanelController&) = delete;
   SidebarWebPanelController& operator=(const SidebarWebPanelController&) =
       delete;
 
-  void OpenWebPanel(const SidebarItem& item);
-  void CloseWebPanel();
-  bool IsShowingWebPanel() const;
+  void ToggleWebPanel(const SidebarItem& item);
+
+  content::WebContents* panel_contents() { return panel_contents_.get(); }
```

Comment:
Can this be const? I see only one place that uses this and it compares pointer.


```suggestion
  const content::WebContents* panel_contents() const { return panel_contents_.get(); }
```

Reply:
good suggestion.